### DISCUSSION TITLE
Open-telemetry data pipline(ETL) use case discussion

### DISCUSSION THREAD

**Discussion by MarsSu0618** [open-telemetry/community#1344] [Community Discussions]
Hi, everyone.
I am a newbie in open-telemetry and as a data engineer in my team.
Because our team would like to introduce this open source to implement observability of microservice. 
Today I want to survey whether it have related data pipeline use case or architecture which can be referred.
By the way, my team's objective is to build AIOps Platform in the future. 
Based on this reason, we need to buildup completed data pipeline to process and store related data so that next step can use ML algorithm to apply in advanced.
If someone have any idea or related experiences, hope you can share it to me. I will be appreciated in advanced.

**Comment by danielbcorreia** [comment]
Hi,
I have a similar use case where I want to add visibility over a data pipeline.
This involves several systems, publication and consumption of messages, and transformations in between.

What I was hoping to get out of something like distributed tracing for this kind of use case is to be able to track the full lifetime of a data object, as well as get information about propagation delays across the whole system.

I'm still trying out a few ideas, but in the end I believe this may be just a visualisation problem - using OpenTelemetry APIs for distributed tracing I believe I can collect all the needed information, it's just that regular tracing UIs are not ready for long lived traces - this use case is a bit different from a client-facing HTTP API. 

There is also the issue of systems that do aggregations - if we think of starting the trace when the first message is produced, and a system downstream needs to aggregate N different messages, we have a problem: we can either publish spans to N different traces, or be able to create a new trace referencing the all original ones. This is still something I need to work out.

**Comment by adirmatzkin** [comment]
Hey! 🤗 @danielbcorreia @MarsSu0618
By any chance you guys had any progress in that area? I was curious about that topic for a while now and wondered if you got to solve this problem or even started to design some sort of solution...

**Reply by danielbcorreia** [reply]
Not yet, but this is something I should be looking at in the next few months.

I've seen people starting to use OTEL to monitor CI/CD pipelines, and it seems to have some similarities to this use case:
https://thenewstack.io/how-to-observe-your-ci-cd-pipelines-with-opentelemetry/

**Reply by adirmatzkin** [reply]
Coming back to this challenge... any news? 🙃 @danielbcorreia

**Reply by trask** [reply]
hi @adirmatzkin, you may find some inspiration in https://github.com/open-telemetry/semantic-conventions/blob/main/docs/messaging/messaging-spans.md
